{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ea434585-0f80-44e5-958d-8ed83a2b8fe5",
   "metadata": {},
   "source": [
    "## Look at the h5 files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e8b6f6dc-980c-44db-8a20-ac4d19b88421",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/orcd/data/edboyden/002/ezh/uni'"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c7aee0ef-0c07-4b98-87d8-5187a48aacf3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Lymphovascular invasion indicator', 'coords', 'features']\n",
      "(3102, 2560)\n"
     ]
    }
   ],
   "source": [
    "import h5py\n",
    "\n",
    "with h5py.File(\"virchow_features/TCGA-A6-2672.h5\", \"r\") as f:\n",
    "    print(list(f.keys()))\n",
    "    print(f[\"features\"].shape)  # (N_patches, 1536)\n",
    "    if \"KRAS_binary\" in f:\n",
    "        print(f[\"KRAS_binary\"][()]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ef9ced6f-1c72-442d-b0cc-533de630c340",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 16\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m     14\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m h5py\u001b[38;5;241m.\u001b[39mFile(path, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n\u001b[1;32m     15\u001b[0m         \u001b[38;5;66;03m# Try \"features\" key first\u001b[39;00m\n\u001b[0;32m---> 16\u001b[0m         data \u001b[38;5;241m=\u001b[39m f[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfeatures\u001b[39m\u001b[38;5;124m\"\u001b[39m][:]\n\u001b[1;32m     17\u001b[0m         \u001b[38;5;66;03m# Squeeze if necessary\u001b[39;00m\n\u001b[1;32m     18\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m data\u001b[38;5;241m.\u001b[39mndim \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m3\u001b[39m:\n",
      "File \u001b[0;32mh5py/_objects.pyx:54\u001b[0m, in \u001b[0;36mh5py._objects.with_phil.wrapper\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mh5py/_objects.pyx:55\u001b[0m, in \u001b[0;36mh5py._objects.with_phil.wrapper\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m/home/software/anaconda3/2023.07/lib/python3.11/site-packages/h5py/_hl/dataset.py:741\u001b[0m, in \u001b[0;36mDataset.__getitem__\u001b[0;34m(self, args, new_dtype)\u001b[0m\n\u001b[1;32m    739\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_fast_read_ok \u001b[38;5;129;01mand\u001b[39;00m (new_dtype \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    740\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 741\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_fast_reader\u001b[38;5;241m.\u001b[39mread(args)\n\u001b[1;32m    742\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m    743\u001b[0m         \u001b[38;5;28;01mpass\u001b[39;00m  \u001b[38;5;66;03m# Fall back to Python read pathway below\u001b[39;00m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import h5py\n",
    "import os\n",
    "from glob import glob\n",
    "\n",
    "# Set the directory where the .h5 files are located\n",
    "h5_folder = \"CLAM/TCGA-COAD/\"  # üîÅ Replace with your actual path\n",
    "h5_paths = sorted(glob(os.path.join(h5_folder, \"**/*.h5\"), recursive=True))\n",
    "\n",
    "# Track patch counts\n",
    "patch_counts = []\n",
    "\n",
    "for path in h5_paths:\n",
    "    try:\n",
    "        with h5py.File(path, \"r\") as f:\n",
    "            # Try \"features\" key first\n",
    "            data = f[\"features\"][:]\n",
    "            # Squeeze if necessary\n",
    "            if data.ndim == 3:\n",
    "                data = data[0]\n",
    "            patch_counts.append((os.path.basename(path), data.shape[0]))\n",
    "    except Exception as e:\n",
    "        patch_counts.append((os.path.basename(path), f\"Error: {e}\"))\n",
    "\n",
    "# Convert to DataFrame for display\n",
    "import pandas as pd\n",
    "import ace_tools as tools\n",
    "\n",
    "patch_df = pd.DataFrame(patch_counts, columns=[\"File\", \"Num_Patches\"])\n",
    "tools.display_dataframe_to_user(name=\"Patch Count per Slide\", dataframe=patch_df)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb12e834-96be-4733-8555-34962110ecb8",
   "metadata": {},
   "source": [
    "Search for nonzero annotations in all the h5 files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "04dfe480-9f72-4e9b-94d3-034a2c157fbb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 0 slides with non-zero annotations.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from glob import glob\n",
    "\n",
    "def has_nonzero_annots(h5_path):\n",
    "    with h5py.File(h5_path, \"r\") as f:\n",
    "        if \"annots\" in f:\n",
    "            annots = f[\"annots\"][:]\n",
    "            return np.any(annots != 0)\n",
    "    return False\n",
    "\n",
    "folder = \"/UNI2-h_features/TCGA-COAD/\"\n",
    "paths = sorted(glob(os.path.join(folder, \"**/*.h5\"), recursive=True))\n",
    "\n",
    "nonzero_annot_slides = [p for p in paths if has_nonzero_annots(p)]\n",
    "\n",
    "print(f\"Found {len(nonzero_annot_slides)} slides with non-zero annotations.\")\n",
    "for path in nonzero_annot_slides:\n",
    "    print(\"found\", os.path.basename(path))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e0d54ef-fc81-45c1-b391-3271f1013d0f",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## Look at TSV file downloaded from CBioPortal for COAD-Read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "09dc8082-ae5a-4c34-8cfb-0e6443f6a9ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded TSV file with shape: (640, 145)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['Study ID',\n",
       " 'Patient ID',\n",
       " 'Sample ID',\n",
       " 'Diagnosis Age',\n",
       " 'American Joint Committee on Cancer Publication Version Type',\n",
       " 'Biopsy Site',\n",
       " 'Cancer Type',\n",
       " 'Cancer Type Detailed',\n",
       " 'Disease Free (Months)',\n",
       " 'Disease Free Status',\n",
       " 'Disease Type',\n",
       " 'Ethnicity Category',\n",
       " 'Fraction Genome Altered',\n",
       " 'ICD-10 Classification',\n",
       " 'Is FFPE',\n",
       " 'Morphology',\n",
       " 'Mutation Count',\n",
       " 'Oncotree Code',\n",
       " 'Overall Survival (Months)',\n",
       " 'Overall Survival Status',\n",
       " 'Other Patient ID',\n",
       " 'Other Sample ID',\n",
       " 'AJCC Pathologic M-Stage',\n",
       " 'AJCC Pathologic N-Stage',\n",
       " 'AJCC Pathologic Stage',\n",
       " 'AJCC Pathologic T-Stage',\n",
       " 'Primary Diagnosis',\n",
       " 'Patient Primary Tumor Site',\n",
       " 'Prior Malignancy',\n",
       " 'Prior Treatment',\n",
       " 'Project Identifier',\n",
       " 'Project Name',\n",
       " 'Project State',\n",
       " 'Race Category',\n",
       " 'Number of Samples Per Patient',\n",
       " 'Sample Type',\n",
       " 'Sex',\n",
       " \"Patient's Vital Status\",\n",
       " 'Year of Death',\n",
       " 'Year of Diagnosis',\n",
       " 'American Joint Committee on Cancer Metastasis Stage Code',\n",
       " 'Neoplasm Disease Lymph Node Stage American Joint Committee on Cancer Code',\n",
       " 'Neoplasm Disease Stage American Joint Committee on Cancer Code',\n",
       " 'American Joint Committee on Cancer Tumor Stage Code',\n",
       " 'BRAF Gene Analysis Indicator',\n",
       " 'BRAF Gene Analysis Result',\n",
       " 'Neoplasm American Joint Committee on Cancer Clinical Distant Metastasis M Stage',\n",
       " 'Neoplasm American Joint Committee on Cancer Clinical Regional Lymph Node N Stage',\n",
       " 'Neoplasm American Joint Committee on Cancer Clinical Primary Tumor T Stage',\n",
       " 'Neoplasm American Joint Committee on Cancer Clinical Group Stage',\n",
       " 'Days to Sample Collection.',\n",
       " 'Last Alive Less Initial Pathologic Diagnosis Date Calculated Day Value',\n",
       " 'days_to_patient_progression_free',\n",
       " 'Days to Sample Procurement',\n",
       " 'days_to_tumor_progression',\n",
       " 'Disease code',\n",
       " 'Lymphomatous Extranodal Site Involvement Indicator',\n",
       " 'Form completion date',\n",
       " 'Patient Height',\n",
       " 'Neoplasm Histologic Type Name',\n",
       " 'Neoadjuvant Therapy Type Administered Prior To Resection Text',\n",
       " 'Prior Cancer Diagnosis Occurence',\n",
       " 'International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code',\n",
       " 'International Classification of Diseases for Oncology, Third Edition ICD-O-3 Site Code',\n",
       " 'Informed consent verified',\n",
       " 'Initial pathologic diagnosis method',\n",
       " 'Year Cancer Initial Diagnosis',\n",
       " 'Kras gene analysis indicator',\n",
       " 'KRAS Mutation',\n",
       " 'Longest Dimension',\n",
       " 'Lymph Node(s) Examined Number',\n",
       " 'Primary Lymph Node Presentation Assessment Ind-3',\n",
       " 'Positive Finding Lymph Node Hematoxylin and Eosin Staining Microscopy Count',\n",
       " 'Positive Finding Lymph Node Keratin Immunohistochemistry Staining Method Count',\n",
       " 'Lymphovascular invasion indicator',\n",
       " 'First Pathologic Diagnosis Biospecimen Acquisition Method Type',\n",
       " 'New Neoplasm Event Post Initial Therapy Indicator',\n",
       " 'Oct embedded',\n",
       " 'Specimen Collection Method',\n",
       " 'Pathology Report File Name',\n",
       " 'Pathology report uuid',\n",
       " 'Perineural Invasion',\n",
       " 'Adjuvant Postoperative Pharmaceutical Therapy Administered Indicator',\n",
       " 'Project code',\n",
       " 'Tissue Prospective Collection Indicator',\n",
       " 'Did patient start adjuvant postoperative radiotherapy?',\n",
       " 'Surgical Margin Resection Status',\n",
       " 'Tissue Retrospective Collection Indicator',\n",
       " 'Sample Initial Weight',\n",
       " 'Sample type id',\n",
       " 'Shortest Dimension',\n",
       " 'Tumor Tissue Site',\n",
       " 'Somatic Status',\n",
       " 'Specimen Current Weight',\n",
       " 'Specimen Freezing Means',\n",
       " 'Specimen Second Longest Dimension',\n",
       " 'Stage Other',\n",
       " 'Time between clamping and freezing',\n",
       " 'Time between excision and freezing',\n",
       " 'Tissue Source Site',\n",
       " 'TMB (nonsynonymous)',\n",
       " 'Primary Therapy Outcome Success Type',\n",
       " 'Person Neoplasm Status',\n",
       " 'Vascular invasion indicator',\n",
       " 'Vial number',\n",
       " 'Patient Weight',\n",
       " 'Aneuploidy Score',\n",
       " 'Buffa Hypoxia Score',\n",
       " 'TCGA PanCanAtlas Cancer Type Acronym',\n",
       " 'Last Communication Contact from Initial Pathologic Diagnosis Date',\n",
       " 'Birth from Initial Pathologic Diagnosis Date',\n",
       " 'Months of disease-specific survival',\n",
       " 'Disease-specific Survival status',\n",
       " 'Genetic Ancestry Label',\n",
       " 'Neoplasm Histologic Grade',\n",
       " 'In PanCan Pathway Analysis',\n",
       " 'MSI MANTIS Score',\n",
       " 'MSIsensor Score',\n",
       " 'Person Neoplasm Cancer Status',\n",
       " 'Progress Free Survival (Months)',\n",
       " 'Progression Free Status',\n",
       " 'Primary Lymph Node Presentation Assessment',\n",
       " 'Prior Diagnosis',\n",
       " 'Radiation Therapy',\n",
       " 'Ragnum Hypoxia Score',\n",
       " 'Subtype',\n",
       " 'Tumor Break Load',\n",
       " 'Tissue Source Site Code',\n",
       " 'Tumor Disease Anatomic Site',\n",
       " 'Tumor Type',\n",
       " 'Winter Hypoxia Score',\n",
       " 'Copy Number',\n",
       " 'Gene Expression Subtype',\n",
       " 'Tumor Other Histologic Subtype',\n",
       " 'Hyper-mutated',\n",
       " 'iCluster',\n",
       " 'Methylation Status',\n",
       " 'Methylation Subtype',\n",
       " 'MLH1 Silencing',\n",
       " 'MRNA Data',\n",
       " 'MSI Status',\n",
       " 'Primary Tumor Site',\n",
       " 'Primary Tumor Pathologic Spread',\n",
       " 'Sequenced',\n",
       " 'Tumor Stage 2009']"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import pprint as pp\n",
    "\n",
    "clinical_path = \"all_tcga_datasets.tsv\"  # or full path if not in current dir\n",
    "df = pd.read_csv(clinical_path, sep=\"\\t\")\n",
    "\n",
    "print(\"Loaded TSV file with shape:\", df.shape)\n",
    "# print(\"\\nColumn names:\")\n",
    "display(df.columns.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "052c09c4-fc61-44d6-bae2-863481873ce0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      NaN\n",
       "1      NaN\n",
       "2      NaN\n",
       "3      NaN\n",
       "4      NaN\n",
       "      ... \n",
       "635    NaN\n",
       "636    NaN\n",
       "637    NaN\n",
       "638    NaN\n",
       "639    YES\n",
       "Name: KRAS Mutation, Length: 640, dtype: object"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(df['KRAS Mutation'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "57aaf819-7bb4-4806-9578-2fbe44186dc3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "KRAS Mutation Value Counts:\n",
      "NaN    623\n",
      "YES     10\n",
      "NO       7\n",
      "Name: KRAS Mutation, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# look at KRAS mutation column\n",
    "if 'KRAS Mutation' in df.columns:\n",
    "    print(\"\\nKRAS Mutation Value Counts:\")\n",
    "    print(df['KRAS Mutation'].value_counts(dropna=False))\n",
    "else:\n",
    "    print(\"\\n'KRAS Mutation' column not found in file.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "584ab051-be30-487c-81f5-7da4c8fb99e2",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## Look at data that might contain MMR status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b0bff420-ee8a-46e5-90ab-e3b0b809e69f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(629, 17)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "['years_to_birth',\n",
       " 'Tumor_purity',\n",
       " 'pathologic_stage',\n",
       " 'pathology_T_stage',\n",
       " 'pathology_N_stage',\n",
       " 'pathology_M_stage',\n",
       " 'histological_type',\n",
       " 'number_of_lymph_nodes',\n",
       " 'Proteomic.subtype',\n",
       " 'gender',\n",
       " 'radiation_therapy',\n",
       " 'residual_tumor',\n",
       " 'ethnicity',\n",
       " 'MSI_phenotype',\n",
       " 'overall_survival',\n",
       " 'status',\n",
       " 'overallsurvival']"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>attrib_name</th>\n",
       "      <th>years_to_birth</th>\n",
       "      <th>Tumor_purity</th>\n",
       "      <th>pathologic_stage</th>\n",
       "      <th>pathology_T_stage</th>\n",
       "      <th>pathology_N_stage</th>\n",
       "      <th>pathology_M_stage</th>\n",
       "      <th>histological_type</th>\n",
       "      <th>number_of_lymph_nodes</th>\n",
       "      <th>Proteomic.subtype</th>\n",
       "      <th>gender</th>\n",
       "      <th>radiation_therapy</th>\n",
       "      <th>residual_tumor</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>MSI_phenotype</th>\n",
       "      <th>overall_survival</th>\n",
       "      <th>status</th>\n",
       "      <th>overallsurvival</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA.A6.2677</th>\n",
       "      <td>68</td>\n",
       "      <td>0.8869</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n2</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>MSS</td>\n",
       "      <td>740</td>\n",
       "      <td>1</td>\n",
       "      <td>740,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.A6.2681</th>\n",
       "      <td>73</td>\n",
       "      <td>0.6939</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>MSS</td>\n",
       "      <td>1387</td>\n",
       "      <td>0</td>\n",
       "      <td>1387,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.A6.2682</th>\n",
       "      <td>70</td>\n",
       "      <td>0.7927</td>\n",
       "      <td>stageiv</td>\n",
       "      <td>t4</td>\n",
       "      <td>n1</td>\n",
       "      <td>m1</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>MSS</td>\n",
       "      <td>424</td>\n",
       "      <td>1</td>\n",
       "      <td>424,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.A6.4105</th>\n",
       "      <td>79</td>\n",
       "      <td>0.7533</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>MSS</td>\n",
       "      <td>442</td>\n",
       "      <td>1</td>\n",
       "      <td>442,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.A6.6649</th>\n",
       "      <td>66</td>\n",
       "      <td>0.7857</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nothispanicorlatino</td>\n",
       "      <td>MSS</td>\n",
       "      <td>735</td>\n",
       "      <td>0</td>\n",
       "      <td>735,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3496</th>\n",
       "      <td>83</td>\n",
       "      <td>0.7004</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>31,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3506</th>\n",
       "      <td>77</td>\n",
       "      <td>0.7979</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>1765</td>\n",
       "      <td>0</td>\n",
       "      <td>1765,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3516</th>\n",
       "      <td>74</td>\n",
       "      <td>0.6843</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n2</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonmucinousadenocarcinoma</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>396</td>\n",
       "      <td>1</td>\n",
       "      <td>396,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3520</th>\n",
       "      <td>86</td>\n",
       "      <td>0.7372</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>731</td>\n",
       "      <td>0</td>\n",
       "      <td>731,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3521</th>\n",
       "      <td>87</td>\n",
       "      <td>0.8137</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA,NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3522</th>\n",
       "      <td>67</td>\n",
       "      <td>0.6299</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>1127</td>\n",
       "      <td>0</td>\n",
       "      <td>1127,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3524</th>\n",
       "      <td>85</td>\n",
       "      <td>0.8229</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>1096</td>\n",
       "      <td>0</td>\n",
       "      <td>1096,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3525</th>\n",
       "      <td>89</td>\n",
       "      <td>0.7697</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n1</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>2</td>\n",
       "      <td>B</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>245</td>\n",
       "      <td>0</td>\n",
       "      <td>245,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3527</th>\n",
       "      <td>89</td>\n",
       "      <td>0.8374</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA,NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3530</th>\n",
       "      <td>80</td>\n",
       "      <td>0.8619</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>580</td>\n",
       "      <td>0</td>\n",
       "      <td>580,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3531</th>\n",
       "      <td>75</td>\n",
       "      <td>0.722</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>1035</td>\n",
       "      <td>0</td>\n",
       "      <td>1035,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3543</th>\n",
       "      <td>84</td>\n",
       "      <td>0.809</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonmucinousadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3552</th>\n",
       "      <td>85</td>\n",
       "      <td>0.7236</td>\n",
       "      <td>stageiii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n2</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>10</td>\n",
       "      <td>D</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>396</td>\n",
       "      <td>1</td>\n",
       "      <td>396,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3554</th>\n",
       "      <td>62</td>\n",
       "      <td>0.5327</td>\n",
       "      <td>stageii</td>\n",
       "      <td>t3</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>colonadenocarcinoma</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>female</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>546</td>\n",
       "      <td>0</td>\n",
       "      <td>546,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.AA.3556</th>\n",
       "      <td>78</td>\n",
       "      <td>0.8469</td>\n",
       "      <td>stagei</td>\n",
       "      <td>t2</td>\n",
       "      <td>n0</td>\n",
       "      <td>m0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>male</td>\n",
       "      <td>no</td>\n",
       "      <td>r0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>700</td>\n",
       "      <td>0</td>\n",
       "      <td>700,0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "attrib_name  years_to_birth Tumor_purity pathologic_stage pathology_T_stage  \\\n",
       "Sample_ID                                                                     \n",
       "TCGA.A6.2677             68       0.8869         stageiii                t3   \n",
       "TCGA.A6.2681             73       0.6939          stageii                t3   \n",
       "TCGA.A6.2682             70       0.7927          stageiv                t4   \n",
       "TCGA.A6.4105             79       0.7533          stageii                t3   \n",
       "TCGA.A6.6649             66       0.7857         stageiii                t3   \n",
       "TCGA.AA.3496             83       0.7004          stageii                t3   \n",
       "TCGA.AA.3506             77       0.7979           stagei                t2   \n",
       "TCGA.AA.3516             74       0.6843         stageiii                t3   \n",
       "TCGA.AA.3520             86       0.7372          stageii                t3   \n",
       "TCGA.AA.3521             87       0.8137          stageii                t3   \n",
       "TCGA.AA.3522             67       0.6299          stageii                t3   \n",
       "TCGA.AA.3524             85       0.8229          stageii                t3   \n",
       "TCGA.AA.3525             89       0.7697         stageiii                t3   \n",
       "TCGA.AA.3527             89       0.8374          stageii                t3   \n",
       "TCGA.AA.3530             80       0.8619           stagei                t2   \n",
       "TCGA.AA.3531             75        0.722          stageii                t3   \n",
       "TCGA.AA.3543             84        0.809           stagei                t2   \n",
       "TCGA.AA.3552             85       0.7236         stageiii                t3   \n",
       "TCGA.AA.3554             62       0.5327          stageii                t3   \n",
       "TCGA.AA.3556             78       0.8469           stagei                t2   \n",
       "\n",
       "attrib_name  pathology_N_stage pathology_M_stage            histological_type  \\\n",
       "Sample_ID                                                                       \n",
       "TCGA.A6.2677                n2                m0          colonadenocarcinoma   \n",
       "TCGA.A6.2681                n0                m0          colonadenocarcinoma   \n",
       "TCGA.A6.2682                n1                m1          colonadenocarcinoma   \n",
       "TCGA.A6.4105                n0                m0          colonadenocarcinoma   \n",
       "TCGA.A6.6649                n1                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3496                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3506                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3516                n2                m0  colonmucinousadenocarcinoma   \n",
       "TCGA.AA.3520                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3521                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3522                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3524                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3525                n1                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3527                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3530                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3531                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3543                n0                m0  colonmucinousadenocarcinoma   \n",
       "TCGA.AA.3552                n2                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3554                n0                m0          colonadenocarcinoma   \n",
       "TCGA.AA.3556                n0                m0                          NaN   \n",
       "\n",
       "attrib_name  number_of_lymph_nodes Proteomic.subtype  gender  \\\n",
       "Sample_ID                                                      \n",
       "TCGA.A6.2677                     5               NaN  female   \n",
       "TCGA.A6.2681                     0               NaN  female   \n",
       "TCGA.A6.2682                     2               NaN    male   \n",
       "TCGA.A6.4105                     0               NaN    male   \n",
       "TCGA.A6.6649                     2               NaN    male   \n",
       "TCGA.AA.3496                     0               NaN  female   \n",
       "TCGA.AA.3506                     0               NaN    male   \n",
       "TCGA.AA.3516                     7               NaN  female   \n",
       "TCGA.AA.3520                     0               NaN  female   \n",
       "TCGA.AA.3521                     0               NaN    male   \n",
       "TCGA.AA.3522                     0               NaN    male   \n",
       "TCGA.AA.3524                     0               NaN    male   \n",
       "TCGA.AA.3525                     2                 B    male   \n",
       "TCGA.AA.3527                     0               NaN  female   \n",
       "TCGA.AA.3530                     0               NaN    male   \n",
       "TCGA.AA.3531                     0                 A  female   \n",
       "TCGA.AA.3543                     0               NaN    male   \n",
       "TCGA.AA.3552                    10                 D    male   \n",
       "TCGA.AA.3554                     0                 C  female   \n",
       "TCGA.AA.3556                     0               NaN    male   \n",
       "\n",
       "attrib_name  radiation_therapy residual_tumor            ethnicity  \\\n",
       "Sample_ID                                                            \n",
       "TCGA.A6.2677                no             r0  nothispanicorlatino   \n",
       "TCGA.A6.2681                no             r0  nothispanicorlatino   \n",
       "TCGA.A6.2682                no             r0  nothispanicorlatino   \n",
       "TCGA.A6.4105                no            NaN  nothispanicorlatino   \n",
       "TCGA.A6.6649                no            NaN  nothispanicorlatino   \n",
       "TCGA.AA.3496               NaN             r0                  NaN   \n",
       "TCGA.AA.3506                no             r0                  NaN   \n",
       "TCGA.AA.3516                no            NaN                  NaN   \n",
       "TCGA.AA.3520                no             r0                  NaN   \n",
       "TCGA.AA.3521               NaN             r0                  NaN   \n",
       "TCGA.AA.3522                no             r0                  NaN   \n",
       "TCGA.AA.3524                no             r0                  NaN   \n",
       "TCGA.AA.3525                no             r0                  NaN   \n",
       "TCGA.AA.3527               NaN             r0                  NaN   \n",
       "TCGA.AA.3530                no             r0                  NaN   \n",
       "TCGA.AA.3531                no             r0                  NaN   \n",
       "TCGA.AA.3543                no             r0                  NaN   \n",
       "TCGA.AA.3552                no             r0                  NaN   \n",
       "TCGA.AA.3554                no             r0                  NaN   \n",
       "TCGA.AA.3556                no             r0                  NaN   \n",
       "\n",
       "attrib_name  MSI_phenotype overall_survival status overallsurvival  \n",
       "Sample_ID                                                           \n",
       "TCGA.A6.2677           MSS              740      1           740,1  \n",
       "TCGA.A6.2681           MSS             1387      0          1387,0  \n",
       "TCGA.A6.2682           MSS              424      1           424,1  \n",
       "TCGA.A6.4105           MSS              442      1           442,1  \n",
       "TCGA.A6.6649           MSS              735      0           735,0  \n",
       "TCGA.AA.3496           MSS               31      0            31,0  \n",
       "TCGA.AA.3506           MSS             1765      0          1765,0  \n",
       "TCGA.AA.3516         MSI-H              396      1           396,1  \n",
       "TCGA.AA.3520           MSS              731      0           731,0  \n",
       "TCGA.AA.3521           MSS              NaN    NaN           NA,NA  \n",
       "TCGA.AA.3522           MSS             1127      0          1127,0  \n",
       "TCGA.AA.3524           MSS             1096      0          1096,0  \n",
       "TCGA.AA.3525         MSI-H              245      0           245,0  \n",
       "TCGA.AA.3527           MSS              NaN    NaN           NA,NA  \n",
       "TCGA.AA.3530           MSS              580      0           580,0  \n",
       "TCGA.AA.3531           MSS             1035      0          1035,0  \n",
       "TCGA.AA.3543         MSI-H               30      0            30,0  \n",
       "TCGA.AA.3552           MSS              396      1           396,1  \n",
       "TCGA.AA.3554         MSI-H              546      0           546,0  \n",
       "TCGA.AA.3556           MSS              700      0           700,0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Step 1: Load the wide format TSV file\n",
    "mmr_path = \"Human__TCGA_COADREAD__MS__Clinical__Clinical__01_28_2016__BI__Clinical__Firehose.tsi.txt\"\n",
    "mmr_df = pd.read_csv(mmr_path, sep=\"\\t\", index_col=0)\n",
    "\n",
    "# Step 2: Transpose so each row is a sample (patient), each column an attribute\n",
    "mmr_df = mmr_df.T\n",
    "mmr_df.index.name = \"Sample_ID\"\n",
    "\n",
    "# Step 3: Preview\n",
    "display(mmr_df.shape)\n",
    "display(mmr_df.columns.tolist())  # column names like 'Mutation_Phenotype'\n",
    "display(mmr_df.head(20))  # first 3 samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f776846c-524a-401a-a419-b33439353d16",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MSS      537\n",
      "MSI-H     89\n",
      "Name: MSI_phenotype, dtype: int64\n",
      "0    537\n",
      "1     89\n",
      "Name: MMR_status, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(mmr_df[\"MSI_phenotype\"].value_counts(dropna=False))\n",
    "\n",
    "# Map MSI-H to 1 (dMMR), MSS to 0 (pMMR)\n",
    "mmr_df[\"MMR_status\"] = mmr_df[\"MSI_phenotype\"].map({\n",
    "    \"MSI-H\": 1,          # dMMR\n",
    "    \"MSS\": 0,            # pMMR\n",
    "})\n",
    "\n",
    "\n",
    "# Drop any samples with NA\n",
    "mmr_df = mmr_df.dropna(subset=[\"MMR_status\"])\n",
    "\n",
    "# Check label distribution\n",
    "print(mmr_df[\"MMR_status\"].value_counts())\n",
    "\n",
    "\n",
    "\n",
    "# display(mmr_df[\"MMR_status\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9f44ab7-5f87-4224-b59e-55425a024be7",
   "metadata": {},
   "source": [
    "## Match each h5 file to label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "86b6fe84-d460-4f00-9a9c-ca15b0c1701b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Lymphovascular invasion indicator</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-3L-AA1B</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-4N-A93T</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-4T-AA8H</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-5M-AAT4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-5M-AAT5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TCGA-5M-AAT6</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TCGA-5M-AATA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TCGA-5M-AATE</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TCGA-A6-2670</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TCGA-A6-2671</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample_ID  Lymphovascular invasion indicator\n",
       "0  TCGA-3L-AA1B                                0.0\n",
       "1  TCGA-4N-A93T                                0.0\n",
       "2  TCGA-4T-AA8H                                0.0\n",
       "3  TCGA-5M-AAT4                                NaN\n",
       "4  TCGA-5M-AAT5                                NaN\n",
       "5  TCGA-5M-AAT6                                1.0\n",
       "6  TCGA-5M-AATA                                NaN\n",
       "7  TCGA-5M-AATE                                0.0\n",
       "8  TCGA-A6-2670                                0.0\n",
       "9  TCGA-A6-2671                                1.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the file\n",
    "mutation_df = pd.read_csv('lymphovascular_invasion.csv')\n",
    "\n",
    "display(mutation_df.head(10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6844aad7-9078-485c-a1a2-b59e6e2330df",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from glob import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ff11e566-25e9-4a5f-a12b-9df52e51e7cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Found 493 .h5 files\n"
     ]
    }
   ],
   "source": [
    "h5_folder = \"virchow_features\"\n",
    "mutation = \"Lymphovascular invasion indicator\"\n",
    "h5_paths = sorted(glob(os.path.join(h5_folder, \"*.h5\")))\n",
    "print(f\"\\nFound {len(h5_paths)} .h5 files\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e4b1207f-21c4-4788-9020-867527a7a746",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "First 5 extracted TCGA IDs:\n",
      "                            h5_path     Sample_ID\n",
      "0  virchow_features/TCGA-3L-AA1B.h5  TCGA-3L-AA1B\n",
      "1  virchow_features/TCGA-4N-A93T.h5  TCGA-4N-A93T\n",
      "2  virchow_features/TCGA-4T-AA8H.h5  TCGA-4T-AA8H\n",
      "3  virchow_features/TCGA-5M-AAT4.h5  TCGA-5M-AAT4\n",
      "4  virchow_features/TCGA-5M-AAT5.h5  TCGA-5M-AAT5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_510306/3986256730.py:11: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  h5_df['Sample_ID'] = h5_df['Sample_ID'].str.replace('.h5', '')\n"
     ]
    }
   ],
   "source": [
    "# Extract TCGA ID from each .h5 filename\n",
    "def extract_tcga_id(path):\n",
    "    basename = os.path.basename(path)\n",
    "    return \"-\".join(basename.split(\"-\")[:3])  # TCGA-XX-YYYY\n",
    "\n",
    "h5_df = pd.DataFrame({\n",
    "    \"h5_path\": h5_paths,\n",
    "    \"Sample_ID\": [extract_tcga_id(p) for p in h5_paths]\n",
    "})\n",
    "\n",
    "h5_df['Sample_ID'] = h5_df['Sample_ID'].str.replace('.h5', '')\n",
    "\n",
    "print(\"\\nFirst 5 extracted TCGA IDs:\")\n",
    "print(h5_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1161e532-1b18-4bc4-9b63-4fe561c8a652",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Sample_ID format in mutation_df:\n",
      "0    TCGA-3L-AA1B\n",
      "1    TCGA-4N-A93T\n",
      "2    TCGA-4T-AA8H\n",
      "3    TCGA-5M-AAT4\n",
      "4    TCGA-5M-AAT5\n",
      "Name: Sample_ID, dtype: object\n",
      "\n",
      "Sample_ID format in h5_df:\n",
      "0    TCGA-3L-AA1B\n",
      "1    TCGA-4N-A93T\n",
      "2    TCGA-4T-AA8H\n",
      "3    TCGA-5M-AAT4\n",
      "4    TCGA-5M-AAT5\n",
      "Name: Sample_ID, dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nSample_ID format in mutation_df:\")\n",
    "print(mutation_df['Sample_ID'].head())\n",
    "\n",
    "print(\"\\nSample_ID format in h5_df:\")\n",
    "print(h5_df['Sample_ID'].head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "5500b28a-99c9-4617-9059-cb83018471ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>h5_path</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Lymphovascular invasion indicator</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>virchow_features/TCGA-3L-AA1B.h5</td>\n",
       "      <td>TCGA-3L-AA1B</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>virchow_features/TCGA-4N-A93T.h5</td>\n",
       "      <td>TCGA-4N-A93T</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>virchow_features/TCGA-4T-AA8H.h5</td>\n",
       "      <td>TCGA-4T-AA8H</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>virchow_features/TCGA-5M-AAT4.h5</td>\n",
       "      <td>TCGA-5M-AAT4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>virchow_features/TCGA-5M-AAT5.h5</td>\n",
       "      <td>TCGA-5M-AAT5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>virchow_features/TCGA-5M-AAT6.h5</td>\n",
       "      <td>TCGA-5M-AAT6</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>virchow_features/TCGA-5M-AATE.h5</td>\n",
       "      <td>TCGA-5M-AATE</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>virchow_features/TCGA-A6-2671.h5</td>\n",
       "      <td>TCGA-A6-2671</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>virchow_features/TCGA-A6-2672.h5</td>\n",
       "      <td>TCGA-A6-2672</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>virchow_features/TCGA-A6-2674.h5</td>\n",
       "      <td>TCGA-A6-2674</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            h5_path     Sample_ID  \\\n",
       "0  virchow_features/TCGA-3L-AA1B.h5  TCGA-3L-AA1B   \n",
       "1  virchow_features/TCGA-4N-A93T.h5  TCGA-4N-A93T   \n",
       "2  virchow_features/TCGA-4T-AA8H.h5  TCGA-4T-AA8H   \n",
       "3  virchow_features/TCGA-5M-AAT4.h5  TCGA-5M-AAT4   \n",
       "4  virchow_features/TCGA-5M-AAT5.h5  TCGA-5M-AAT5   \n",
       "5  virchow_features/TCGA-5M-AAT6.h5  TCGA-5M-AAT6   \n",
       "6  virchow_features/TCGA-5M-AATE.h5  TCGA-5M-AATE   \n",
       "7  virchow_features/TCGA-A6-2671.h5  TCGA-A6-2671   \n",
       "8  virchow_features/TCGA-A6-2672.h5  TCGA-A6-2672   \n",
       "9  virchow_features/TCGA-A6-2674.h5  TCGA-A6-2674   \n",
       "\n",
       "   Lymphovascular invasion indicator  \n",
       "0                                0.0  \n",
       "1                                0.0  \n",
       "2                                0.0  \n",
       "3                                NaN  \n",
       "4                                NaN  \n",
       "5                                1.0  \n",
       "6                                0.0  \n",
       "7                                1.0  \n",
       "8                                1.0  \n",
       "9                                0.0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Matched: 456\n",
      "Unmatched: 37\n",
      "\n",
      "Matched sample preview:\n",
      "      Sample_ID  Lymphovascular invasion indicator\n",
      "0  TCGA-3L-AA1B                                0.0\n",
      "1  TCGA-4N-A93T                                0.0\n",
      "2  TCGA-4T-AA8H                                0.0\n",
      "5  TCGA-5M-AAT6                                1.0\n",
      "6  TCGA-5M-AATE                                0.0\n"
     ]
    }
   ],
   "source": [
    "merged_df = h5_df.merge(mutation_df, on=\"Sample_ID\", how=\"left\")\n",
    "\n",
    "display(merged_df.head(10))\n",
    "\n",
    "# Separate matched/unmatched again\n",
    "matched = merged_df.dropna(subset=[mutation])\n",
    "unmatched = merged_df[merged_df[mutation].isna()]\n",
    "\n",
    "# Print results\n",
    "print(f\"\\nMatched: {len(matched)}\")\n",
    "print(f\"Unmatched: {len(unmatched)}\")\n",
    "print(\"\\nMatched sample preview:\")\n",
    "print(matched[[\"Sample_ID\", mutation]].head(5))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a9d2a49f-d07a-40ee-8868-3ddf55f12595",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0    281\n",
      "1.0    175\n",
      "Name: Lymphovascular invasion indicator, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(matched[mutation].value_counts())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef26c5b8-003e-4432-b5cd-ea860e7daf05",
   "metadata": {},
   "source": [
    "### Save the mutation status column in the h5 file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "2cde4e3e-403e-4933-8a19-74c734eac694",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing Lymphovascular invasion indicator to TCGA-3L-AA1B.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-4N-A93T.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-4T-AA8H.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-5M-AAT6.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-5M-AATE.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2671.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2672.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2674.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2675.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2676.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2677.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2678.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2679.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2680.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2681.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2682.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2683.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2684.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2685.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-2686.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-3807.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-3808.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-3809.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-3810.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-4105.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-4107.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5656.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5657.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5659.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5660.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5661.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5662.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5664.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5665.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5666.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-5667.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6137.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6138.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6140.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6141.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6142.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6648.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6649.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6650.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6651.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6652.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6653.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-6654.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-A565.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-A566.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-A567.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-A56B.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-A6-A5ZU.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3492.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3509.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3510.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3511.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3514.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3516.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3517.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3518.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3519.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3520.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3521.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3522.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3524.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3525.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3526.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3527.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3529.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3530.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3531.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3532.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3534.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3538.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3542.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3543.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3544.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3548.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3549.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3552.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3553.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3554.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3555.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3556.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3558.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3560.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3561.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3562.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3655.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3660.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3662.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3663.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3664.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3666.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3667.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3672.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3673.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3675.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3678.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3679.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3680.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3681.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3684.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3685.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3688.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3692.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3693.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3695.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3696.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3710.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3712.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3715.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3811.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3812.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3814.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3815.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3818.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3819.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3831.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3833.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3837.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3841.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3842.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3844.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3845.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3846.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3848.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3850.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3851.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3852.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3854.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3855.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3856.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3858.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3860.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3861.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3862.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3864.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3866.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3867.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3869.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3870.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3872.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3875.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3877.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3930.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3939.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3941.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3947.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3949.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3950.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3952.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3955.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3956.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3966.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3967.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3968.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3970.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3971.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3972.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3973.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3975.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3977.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3979.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3980.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3982.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3984.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3986.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3989.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-3994.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A004.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00A.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00D.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00E.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00F.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00J.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00K.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00L.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00N.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00O.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00Q.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00R.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00U.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00W.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A00Z.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A010.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A017.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01C.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01D.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01F.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01G.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01I.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01K.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01P.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01Q.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01R.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01T.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01X.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A01Z.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A022.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02E.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02F.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02H.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02J.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02K.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02O.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02R.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AA-A02Y.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-5900.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6548.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6888.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6889.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6890.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6895.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6899.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6901.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6963.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6964.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-6965.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AD-A5EJ.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-2687.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-2690.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-2693.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-3911.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-4110.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-5654.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-6136.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-6655.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-6672.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-A56K.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-A56L.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AF-A56N.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-3591.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-3592.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-3725.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-3732.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-3902.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-4021.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AG-4022.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AH-6547.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AH-6643.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AH-6644.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AH-6897.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AH-6903.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AM-5820.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AM-5821.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AU-3779.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AU-6004.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-4070.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-4071.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-5543.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-6196.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-6197.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-6386.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-A54L.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-A69D.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-A71X.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AY-A8YK.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4308.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4313.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4315.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4614.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4615.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4616.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4682.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-AZ-4684.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-4952.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-5912.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-5913.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-5914.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-5915.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-5916.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-6746.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-6747.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-6748.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CK-6751.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CL-5917.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CL-5918.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4743.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4744.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4746.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4747.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4748.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4750.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4751.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-4752.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5341.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5344.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5348.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5349.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5860.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5861.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5862.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5863.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5864.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-5868.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6161.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6162.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6163.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6164.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6165.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6166.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6167.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6168.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6169.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6170.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6171.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6172.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6674.h5\n",
      "Error writing to virchow_features/TCGA-CM-6675.h5: [Errno 11] Unable to open file (unable to lock file, errno = 11, error message = 'Resource temporarily unavailable')\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6676.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6677.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6678.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6679.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-CM-6680.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-5537.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-5538.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-5539.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-5540.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-5541.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6529.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6530.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6531.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6532.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6533.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6534.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6535.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6536.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6537.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6538.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6539.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6540.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6541.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6898.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6920.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6922.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6923.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6924.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6926.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6927.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6928.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6929.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6930.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6931.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-6932.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-D5-7000.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-4745.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-4749.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-5337.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-5869.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6154.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6155.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6156.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6157.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6158.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6160.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6681.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6682.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DC-6683.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A0X9.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A0XD.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A0XF.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A1D0.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A1D4.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A1D6.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A1D7.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A1D9.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A1DB.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A280.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A282.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A288.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A28A.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A28E.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A28G.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A28H.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A28K.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DM-A28M.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DY-A0XA.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DY-A1DC.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-DY-A1DE.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EF-5830.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6506.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6507.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6882.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6883.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6884.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6885.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-6917.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-7002.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-EI-7004.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6459.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6460.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6461.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6463.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6569.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6570.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6703.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6704.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6805.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6806.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6807.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6808.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6809.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6854.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6855.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F4-6856.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6464.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6465.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6571.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6702.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6811.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6812.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6813.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6814.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6863.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-F5-6864.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6293.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6294.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6295.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6297.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6298.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6299.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6302.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6303.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6320.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6321.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6322.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6323.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6586.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6588.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6625.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6626.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6627.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G4-6628.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G5-6233.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G5-6235.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G5-6572.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-G5-6641.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A50T.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A50U.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A50V.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A5IV.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A6GA.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A6GB.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A6GC.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A8F7.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-NH-A8F8.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-QG-A5YV.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-QG-A5YW.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-QG-A5YX.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-QG-A5Z1.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-QG-A5Z2.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-SS-A7HO.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-T9-A92H.h5\n",
      "Writing Lymphovascular invasion indicator to TCGA-WS-AB45.h5\n"
     ]
    }
   ],
   "source": [
    "import h5py\n",
    "\n",
    "# Loop over matched rows\n",
    "for idx, row in matched.iterrows():\n",
    "    h5_path = row[\"h5_path\"]\n",
    "    label = int(row[mutation])  # make sure it's a simple 0 or 1\n",
    "    \n",
    "    try:\n",
    "        with h5py.File(h5_path, \"a\") as f:  # 'a' = append mode\n",
    "            if mutation in f:\n",
    "                print(f\"Overwriting existing {mutation} in {os.path.basename(h5_path)}\")\n",
    "                del f[mutation]\n",
    "            else:\n",
    "                print(f\"Writing {mutation} to {os.path.basename(h5_path)}\")\n",
    "            \n",
    "            f.create_dataset(mutation, data=label)\n",
    "    \n",
    "    except Exception as e:\n",
    "        print(f\"Error writing to {h5_path}: {e}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "0c6408e4-7f6f-4474-8f58-5d750ac1a479",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['h5_path', 'Sample_ID', 'Lymphovascular invasion indicator'], dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>h5_path</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Lymphovascular invasion indicator</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>virchow_features/TCGA-3L-AA1B.h5</td>\n",
       "      <td>TCGA-3L-AA1B</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>virchow_features/TCGA-4N-A93T.h5</td>\n",
       "      <td>TCGA-4N-A93T</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>virchow_features/TCGA-4T-AA8H.h5</td>\n",
       "      <td>TCGA-4T-AA8H</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>virchow_features/TCGA-5M-AAT6.h5</td>\n",
       "      <td>TCGA-5M-AAT6</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>virchow_features/TCGA-5M-AATE.h5</td>\n",
       "      <td>TCGA-5M-AATE</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>487</th>\n",
       "      <td>virchow_features/TCGA-QG-A5Z1.h5</td>\n",
       "      <td>TCGA-QG-A5Z1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>488</th>\n",
       "      <td>virchow_features/TCGA-QG-A5Z2.h5</td>\n",
       "      <td>TCGA-QG-A5Z2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>virchow_features/TCGA-SS-A7HO.h5</td>\n",
       "      <td>TCGA-SS-A7HO</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>virchow_features/TCGA-T9-A92H.h5</td>\n",
       "      <td>TCGA-T9-A92H</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>virchow_features/TCGA-WS-AB45.h5</td>\n",
       "      <td>TCGA-WS-AB45</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>456 rows √ó 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              h5_path     Sample_ID  \\\n",
       "0    virchow_features/TCGA-3L-AA1B.h5  TCGA-3L-AA1B   \n",
       "1    virchow_features/TCGA-4N-A93T.h5  TCGA-4N-A93T   \n",
       "2    virchow_features/TCGA-4T-AA8H.h5  TCGA-4T-AA8H   \n",
       "5    virchow_features/TCGA-5M-AAT6.h5  TCGA-5M-AAT6   \n",
       "6    virchow_features/TCGA-5M-AATE.h5  TCGA-5M-AATE   \n",
       "..                                ...           ...   \n",
       "487  virchow_features/TCGA-QG-A5Z1.h5  TCGA-QG-A5Z1   \n",
       "488  virchow_features/TCGA-QG-A5Z2.h5  TCGA-QG-A5Z2   \n",
       "490  virchow_features/TCGA-SS-A7HO.h5  TCGA-SS-A7HO   \n",
       "491  virchow_features/TCGA-T9-A92H.h5  TCGA-T9-A92H   \n",
       "492  virchow_features/TCGA-WS-AB45.h5  TCGA-WS-AB45   \n",
       "\n",
       "     Lymphovascular invasion indicator  \n",
       "0                                  0.0  \n",
       "1                                  0.0  \n",
       "2                                  0.0  \n",
       "5                                  1.0  \n",
       "6                                  0.0  \n",
       "..                                 ...  \n",
       "487                                1.0  \n",
       "488                                0.0  \n",
       "490                                1.0  \n",
       "491                                0.0  \n",
       "492                                0.0  \n",
       "\n",
       "[456 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(matched.keys())\n",
    "display(matched)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9206a4a4-f68d-48fa-a758-79415f113349",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-3L-AA1B.h5, 0\n",
      "TCGA-4N-A93T.h5, 0\n",
      "TCGA-4T-AA8H.h5, 0\n",
      "TCGA-5M-AAT4.h5 missing mutation\n",
      "TCGA-5M-AAT5.h5 missing mutation\n",
      "TCGA-5M-AAT6.h5, 1\n",
      "TCGA-5M-AATE.h5, 0\n",
      "TCGA-A6-2671.h5, 1\n",
      "TCGA-A6-2672.h5, 1\n",
      "TCGA-A6-2674.h5, 0\n",
      "TCGA-A6-2675.h5, 0\n",
      "TCGA-A6-2676.h5, 0\n",
      "TCGA-A6-2677.h5, 1\n",
      "TCGA-A6-2678.h5, 1\n",
      "TCGA-A6-2679.h5, 0\n",
      "TCGA-A6-2680.h5, 0\n",
      "TCGA-A6-2681.h5, 0\n",
      "TCGA-A6-2682.h5, 1\n",
      "TCGA-A6-2683.h5, 0\n",
      "TCGA-A6-2684.h5, 0\n",
      "TCGA-A6-2685.h5, 0\n",
      "TCGA-A6-2686.h5, 0\n",
      "TCGA-A6-3807.h5, 1\n",
      "TCGA-A6-3808.h5, 0\n",
      "TCGA-A6-3809.h5, 0\n",
      "TCGA-A6-3810.h5, 0\n",
      "TCGA-A6-4105.h5, 0\n",
      "TCGA-A6-4107.h5, 1\n",
      "TCGA-A6-5656.h5, 0\n",
      "TCGA-A6-5657.h5, 1\n",
      "TCGA-A6-5659.h5, 0\n",
      "TCGA-A6-5660.h5, 1\n",
      "TCGA-A6-5661.h5, 0\n",
      "TCGA-A6-5662.h5, 1\n",
      "TCGA-A6-5664.h5, 1\n",
      "TCGA-A6-5665.h5, 1\n",
      "TCGA-A6-5666.h5, 0\n",
      "TCGA-A6-5667.h5, 1\n",
      "TCGA-A6-6137.h5, 0\n",
      "TCGA-A6-6138.h5, 0\n",
      "TCGA-A6-6140.h5, 0\n",
      "TCGA-A6-6141.h5, 0\n",
      "TCGA-A6-6142.h5, 1\n",
      "TCGA-A6-6648.h5, 0\n",
      "TCGA-A6-6649.h5, 1\n",
      "TCGA-A6-6650.h5, 0\n",
      "TCGA-A6-6651.h5, 1\n",
      "TCGA-A6-6652.h5, 0\n",
      "TCGA-A6-6653.h5, 0\n",
      "TCGA-A6-6654.h5, 1\n",
      "TCGA-A6-A565.h5, 1\n",
      "TCGA-A6-A566.h5, 1\n",
      "TCGA-A6-A567.h5, 1\n",
      "TCGA-A6-A56B.h5, 1\n",
      "TCGA-A6-A5ZU.h5, 1\n",
      "TCGA-AA-3488.h5 missing mutation\n",
      "TCGA-AA-3489.h5 missing mutation\n",
      "TCGA-AA-3492.h5, 0\n",
      "TCGA-AA-3494.h5 missing mutation\n",
      "TCGA-AA-3495.h5 missing mutation\n",
      "TCGA-AA-3496.h5 missing mutation\n",
      "TCGA-AA-3506.h5 missing mutation\n",
      "TCGA-AA-3509.h5, 0\n",
      "TCGA-AA-3510.h5, 0\n",
      "TCGA-AA-3511.h5, 1\n",
      "TCGA-AA-3514.h5, 0\n",
      "TCGA-AA-3516.h5, 0\n",
      "TCGA-AA-3517.h5, 0\n",
      "TCGA-AA-3518.h5, 0\n",
      "TCGA-AA-3519.h5, 0\n",
      "TCGA-AA-3520.h5, 0\n",
      "TCGA-AA-3521.h5, 0\n",
      "TCGA-AA-3522.h5, 1\n",
      "TCGA-AA-3524.h5, 0\n",
      "TCGA-AA-3525.h5, 0\n",
      "TCGA-AA-3526.h5, 0\n",
      "TCGA-AA-3527.h5, 0\n",
      "TCGA-AA-3529.h5, 1\n",
      "TCGA-AA-3530.h5, 0\n",
      "TCGA-AA-3531.h5, 0\n",
      "TCGA-AA-3532.h5, 0\n",
      "TCGA-AA-3534.h5, 1\n",
      "TCGA-AA-3538.h5, 0\n",
      "TCGA-AA-3542.h5, 1\n",
      "TCGA-AA-3543.h5, 0\n",
      "TCGA-AA-3544.h5, 0\n",
      "TCGA-AA-3548.h5, 1\n",
      "TCGA-AA-3549.h5, 0\n",
      "TCGA-AA-3552.h5, 1\n",
      "TCGA-AA-3553.h5, 0\n",
      "TCGA-AA-3554.h5, 0\n",
      "TCGA-AA-3555.h5, 0\n",
      "TCGA-AA-3556.h5, 0\n",
      "TCGA-AA-3558.h5, 0\n",
      "TCGA-AA-3560.h5, 1\n",
      "TCGA-AA-3561.h5, 0\n",
      "TCGA-AA-3562.h5, 1\n",
      "TCGA-AA-3655.h5, 0\n",
      "TCGA-AA-3660.h5, 0\n",
      "TCGA-AA-3662.h5, 0\n",
      "TCGA-AA-3663.h5, 0\n",
      "TCGA-AA-3664.h5, 0\n",
      "TCGA-AA-3666.h5, 0\n",
      "TCGA-AA-3667.h5, 0\n",
      "TCGA-AA-3672.h5, 0\n",
      "TCGA-AA-3673.h5, 0\n",
      "TCGA-AA-3675.h5, 0\n",
      "TCGA-AA-3678.h5, 1\n",
      "TCGA-AA-3679.h5, 1\n",
      "TCGA-AA-3680.h5, 1\n",
      "TCGA-AA-3681.h5, 0\n",
      "TCGA-AA-3684.h5, 1\n",
      "TCGA-AA-3685.h5, 1\n",
      "TCGA-AA-3688.h5, 1\n",
      "TCGA-AA-3692.h5, 0\n",
      "TCGA-AA-3693.h5, 1\n",
      "TCGA-AA-3695.h5, 1\n",
      "TCGA-AA-3696.h5, 0\n",
      "TCGA-AA-3697.h5 missing mutation\n",
      "TCGA-AA-3710.h5, 1\n",
      "TCGA-AA-3712.h5, 1\n",
      "TCGA-AA-3713.h5 missing mutation\n",
      "TCGA-AA-3715.h5, 1\n",
      "TCGA-AA-3811.h5, 1\n",
      "TCGA-AA-3812.h5, 1\n",
      "TCGA-AA-3814.h5, 1\n",
      "TCGA-AA-3815.h5, 1\n",
      "TCGA-AA-3818.h5, 0\n",
      "TCGA-AA-3819.h5, 1\n",
      "TCGA-AA-3821.h5 missing mutation\n",
      "TCGA-AA-3831.h5, 1\n",
      "TCGA-AA-3833.h5, 1\n",
      "TCGA-AA-3837.h5, 1\n",
      "TCGA-AA-3841.h5, 1\n",
      "TCGA-AA-3842.h5, 1\n",
      "TCGA-AA-3844.h5, 1\n",
      "TCGA-AA-3845.h5, 1\n",
      "TCGA-AA-3846.h5, 1\n",
      "TCGA-AA-3848.h5, 1\n",
      "TCGA-AA-3850.h5, 1\n",
      "TCGA-AA-3851.h5, 1\n",
      "TCGA-AA-3852.h5, 1\n",
      "TCGA-AA-3854.h5, 1\n",
      "TCGA-AA-3855.h5, 1\n",
      "TCGA-AA-3856.h5, 1\n",
      "TCGA-AA-3858.h5, 1\n",
      "TCGA-AA-3860.h5, 1\n",
      "TCGA-AA-3861.h5, 1\n",
      "TCGA-AA-3862.h5, 1\n",
      "TCGA-AA-3864.h5, 0\n",
      "TCGA-AA-3866.h5, 0\n",
      "TCGA-AA-3867.h5, 1\n",
      "TCGA-AA-3869.h5, 1\n",
      "TCGA-AA-3870.h5, 1\n",
      "TCGA-AA-3872.h5, 1\n",
      "TCGA-AA-3875.h5, 0\n",
      "TCGA-AA-3877.h5, 0\n",
      "TCGA-AA-3930.h5, 1\n",
      "TCGA-AA-3939.h5, 0\n",
      "TCGA-AA-3941.h5, 1\n",
      "TCGA-AA-3947.h5, 0\n",
      "TCGA-AA-3949.h5, 0\n",
      "TCGA-AA-3950.h5, 0\n",
      "TCGA-AA-3952.h5, 1\n",
      "TCGA-AA-3955.h5, 1\n",
      "TCGA-AA-3956.h5, 0\n",
      "TCGA-AA-3966.h5, 1\n",
      "TCGA-AA-3967.h5, 0\n",
      "TCGA-AA-3968.h5, 0\n",
      "TCGA-AA-3970.h5, 0\n",
      "TCGA-AA-3971.h5, 0\n",
      "TCGA-AA-3972.h5, 0\n",
      "TCGA-AA-3973.h5, 1\n",
      "TCGA-AA-3975.h5, 0\n",
      "TCGA-AA-3976.h5 missing mutation\n",
      "TCGA-AA-3977.h5, 0\n",
      "TCGA-AA-3979.h5, 0\n",
      "TCGA-AA-3980.h5, 1\n",
      "TCGA-AA-3982.h5, 1\n",
      "TCGA-AA-3984.h5, 1\n",
      "TCGA-AA-3986.h5, 0\n",
      "TCGA-AA-3989.h5, 1\n",
      "TCGA-AA-3994.h5, 0\n",
      "TCGA-AA-A004.h5, 0\n",
      "TCGA-AA-A00A.h5, 0\n",
      "TCGA-AA-A00D.h5, 0\n",
      "TCGA-AA-A00E.h5, 0\n",
      "TCGA-AA-A00F.h5, 1\n",
      "TCGA-AA-A00J.h5, 1\n",
      "TCGA-AA-A00K.h5, 0\n",
      "TCGA-AA-A00L.h5, 0\n",
      "TCGA-AA-A00N.h5, 1\n",
      "TCGA-AA-A00O.h5, 1\n",
      "TCGA-AA-A00Q.h5, 1\n",
      "TCGA-AA-A00R.h5, 0\n",
      "TCGA-AA-A00U.h5, 0\n",
      "TCGA-AA-A00W.h5, 0\n",
      "TCGA-AA-A00Z.h5, 1\n",
      "TCGA-AA-A010.h5, 0\n",
      "TCGA-AA-A017.h5, 1\n",
      "TCGA-AA-A01C.h5, 1\n",
      "TCGA-AA-A01D.h5, 1\n",
      "TCGA-AA-A01F.h5, 1\n",
      "TCGA-AA-A01G.h5, 0\n",
      "TCGA-AA-A01I.h5, 0\n",
      "TCGA-AA-A01K.h5, 1\n",
      "TCGA-AA-A01P.h5, 1\n",
      "TCGA-AA-A01Q.h5, 1\n",
      "TCGA-AA-A01R.h5, 1\n",
      "TCGA-AA-A01S.h5 missing mutation\n",
      "TCGA-AA-A01T.h5, 1\n",
      "TCGA-AA-A01V.h5 missing mutation\n",
      "TCGA-AA-A01X.h5, 1\n",
      "TCGA-AA-A01Z.h5, 1\n",
      "TCGA-AA-A022.h5, 1\n",
      "TCGA-AA-A024.h5 missing mutation\n",
      "TCGA-AA-A029.h5 missing mutation\n",
      "TCGA-AA-A02E.h5, 1\n",
      "TCGA-AA-A02F.h5, 1\n",
      "TCGA-AA-A02H.h5, 1\n",
      "TCGA-AA-A02J.h5, 1\n",
      "TCGA-AA-A02K.h5, 1\n",
      "TCGA-AA-A02O.h5, 1\n",
      "TCGA-AA-A02R.h5, 1\n",
      "TCGA-AA-A02W.h5 missing mutation\n",
      "TCGA-AA-A02Y.h5, 1\n",
      "TCGA-AA-A03F.h5 missing mutation\n",
      "TCGA-AA-A03J.h5 missing mutation\n",
      "TCGA-AD-5900.h5, 0\n",
      "TCGA-AD-6548.h5, 0\n",
      "TCGA-AD-6888.h5, 1\n",
      "TCGA-AD-6889.h5, 0\n",
      "TCGA-AD-6890.h5, 0\n",
      "TCGA-AD-6895.h5, 1\n",
      "TCGA-AD-6899.h5, 1\n",
      "TCGA-AD-6901.h5, 0\n",
      "TCGA-AD-6963.h5, 0\n",
      "TCGA-AD-6964.h5, 1\n",
      "TCGA-AD-6965.h5, 1\n",
      "TCGA-AD-A5EJ.h5, 0\n",
      "TCGA-AD-A5EK.h5 missing mutation\n",
      "TCGA-AF-2687.h5, 1\n",
      "TCGA-AF-2690.h5, 1\n",
      "TCGA-AF-2693.h5, 0\n",
      "TCGA-AF-3911.h5, 1\n",
      "TCGA-AF-4110.h5, 1\n",
      "TCGA-AF-5654.h5, 0\n",
      "TCGA-AF-6136.h5, 1\n",
      "TCGA-AF-6655.h5, 1\n",
      "TCGA-AF-6672.h5, 1\n",
      "TCGA-AF-A56K.h5, 0\n",
      "TCGA-AF-A56L.h5, 1\n",
      "TCGA-AF-A56N.h5, 0\n",
      "TCGA-AG-3591.h5, 0\n",
      "TCGA-AG-3592.h5, 0\n",
      "TCGA-AG-3725.h5, 1\n",
      "TCGA-AG-3731.h5 missing mutation\n",
      "TCGA-AG-3732.h5, 1\n",
      "TCGA-AG-3742.h5 missing mutation\n",
      "TCGA-AG-3902.h5, 0\n",
      "TCGA-AG-4021.h5, 1\n",
      "TCGA-AG-4022.h5, 1\n",
      "TCGA-AH-6544.h5 missing mutation\n",
      "TCGA-AH-6547.h5, 0\n",
      "TCGA-AH-6643.h5, 0\n",
      "TCGA-AH-6644.h5, 0\n",
      "TCGA-AH-6897.h5, 0\n",
      "TCGA-AH-6903.h5, 0\n",
      "TCGA-AM-5820.h5, 1\n",
      "TCGA-AM-5821.h5, 0\n",
      "TCGA-AU-3779.h5, 0\n",
      "TCGA-AU-6004.h5, 0\n",
      "TCGA-AY-4070.h5, 1\n",
      "TCGA-AY-4071.h5, 0\n",
      "TCGA-AY-5543.h5, 0\n",
      "TCGA-AY-6196.h5, 1\n",
      "TCGA-AY-6197.h5, 0\n",
      "TCGA-AY-6386.h5, 0\n",
      "TCGA-AY-A54L.h5, 0\n",
      "TCGA-AY-A69D.h5, 0\n",
      "TCGA-AY-A71X.h5, 0\n",
      "TCGA-AY-A8YK.h5, 1\n",
      "TCGA-AZ-4308.h5, 1\n",
      "TCGA-AZ-4313.h5, 1\n",
      "TCGA-AZ-4315.h5, 0\n",
      "TCGA-AZ-4614.h5, 1\n",
      "TCGA-AZ-4615.h5, 1\n",
      "TCGA-AZ-4616.h5, 1\n",
      "TCGA-AZ-4681.h5 missing mutation\n",
      "TCGA-AZ-4682.h5, 0\n",
      "TCGA-AZ-4684.h5, 1\n",
      "TCGA-CK-4952.h5, 0\n",
      "TCGA-CK-5912.h5, 0\n",
      "TCGA-CK-5913.h5, 0\n",
      "TCGA-CK-5914.h5, 0\n",
      "TCGA-CK-5915.h5, 0\n",
      "TCGA-CK-5916.h5, 1\n",
      "TCGA-CK-6746.h5, 0\n",
      "TCGA-CK-6747.h5, 0\n",
      "TCGA-CK-6748.h5, 0\n",
      "TCGA-CK-6751.h5, 0\n",
      "TCGA-CL-4957.h5 missing mutation\n",
      "TCGA-CL-5917.h5, 1\n",
      "TCGA-CL-5918.h5, 0\n",
      "TCGA-CM-4743.h5, 1\n",
      "TCGA-CM-4744.h5, 1\n",
      "TCGA-CM-4746.h5, 0\n",
      "TCGA-CM-4747.h5, 1\n",
      "TCGA-CM-4748.h5, 1\n",
      "TCGA-CM-4750.h5, 1\n",
      "TCGA-CM-4751.h5, 1\n",
      "TCGA-CM-4752.h5, 0\n",
      "TCGA-CM-5341.h5, 1\n",
      "TCGA-CM-5344.h5, 1\n",
      "TCGA-CM-5348.h5, 1\n",
      "TCGA-CM-5349.h5, 0\n",
      "TCGA-CM-5860.h5, 0\n",
      "TCGA-CM-5861.h5, 0\n",
      "TCGA-CM-5862.h5, 0\n",
      "TCGA-CM-5863.h5, 0\n",
      "TCGA-CM-5864.h5, 0\n",
      "TCGA-CM-5868.h5, 1\n",
      "TCGA-CM-6161.h5, 0\n",
      "TCGA-CM-6162.h5, 1\n",
      "TCGA-CM-6163.h5, 0\n",
      "TCGA-CM-6164.h5, 0\n",
      "TCGA-CM-6165.h5, 0\n",
      "TCGA-CM-6166.h5, 0\n",
      "TCGA-CM-6167.h5, 1\n",
      "TCGA-CM-6168.h5, 0\n",
      "TCGA-CM-6169.h5, 1\n",
      "TCGA-CM-6170.h5, 0\n",
      "TCGA-CM-6171.h5, 0\n",
      "TCGA-CM-6172.h5, 0\n",
      "TCGA-CM-6674.h5, 0\n",
      "TCGA-CM-6675.h5 missing mutation\n",
      "TCGA-CM-6676.h5, 1\n",
      "TCGA-CM-6677.h5, 0\n",
      "TCGA-CM-6678.h5, 0\n",
      "TCGA-CM-6679.h5, 0\n",
      "TCGA-CM-6680.h5, 0\n",
      "TCGA-D5-5537.h5, 0\n",
      "TCGA-D5-5538.h5, 0\n",
      "TCGA-D5-5539.h5, 1\n",
      "TCGA-D5-5540.h5, 0\n",
      "TCGA-D5-5541.h5, 0\n",
      "TCGA-D5-6529.h5, 0\n",
      "TCGA-D5-6530.h5, 0\n",
      "TCGA-D5-6531.h5, 0\n",
      "TCGA-D5-6532.h5, 0\n",
      "TCGA-D5-6533.h5, 0\n",
      "TCGA-D5-6534.h5, 0\n",
      "TCGA-D5-6535.h5, 1\n",
      "TCGA-D5-6536.h5, 0\n",
      "TCGA-D5-6537.h5, 1\n",
      "TCGA-D5-6538.h5, 0\n",
      "TCGA-D5-6539.h5, 0\n",
      "TCGA-D5-6540.h5, 0\n",
      "TCGA-D5-6541.h5, 0\n",
      "TCGA-D5-6898.h5, 0\n",
      "TCGA-D5-6920.h5, 0\n",
      "TCGA-D5-6922.h5, 1\n",
      "TCGA-D5-6923.h5, 0\n",
      "TCGA-D5-6924.h5, 0\n",
      "TCGA-D5-6926.h5, 0\n",
      "TCGA-D5-6927.h5, 0\n",
      "TCGA-D5-6928.h5, 0\n",
      "TCGA-D5-6929.h5, 1\n",
      "TCGA-D5-6930.h5, 0\n",
      "TCGA-D5-6931.h5, 1\n",
      "TCGA-D5-6932.h5, 0\n",
      "TCGA-D5-7000.h5, 0\n",
      "TCGA-DC-4745.h5, 0\n",
      "TCGA-DC-4749.h5, 1\n",
      "TCGA-DC-5337.h5, 0\n",
      "TCGA-DC-5869.h5, 0\n",
      "TCGA-DC-6154.h5, 1\n",
      "TCGA-DC-6155.h5, 0\n",
      "TCGA-DC-6156.h5, 1\n",
      "TCGA-DC-6157.h5, 0\n",
      "TCGA-DC-6158.h5, 0\n",
      "TCGA-DC-6160.h5, 0\n",
      "TCGA-DC-6681.h5, 1\n",
      "TCGA-DC-6682.h5, 0\n",
      "TCGA-DC-6683.h5, 0\n",
      "TCGA-DM-A0X9.h5, 0\n",
      "TCGA-DM-A0XD.h5, 0\n",
      "TCGA-DM-A0XF.h5, 0\n",
      "TCGA-DM-A1D0.h5, 0\n",
      "TCGA-DM-A1D4.h5, 0\n",
      "TCGA-DM-A1D6.h5, 0\n",
      "TCGA-DM-A1D7.h5, 0\n",
      "TCGA-DM-A1D8.h5 missing mutation\n",
      "TCGA-DM-A1D9.h5, 0\n",
      "TCGA-DM-A1DA.h5 missing mutation\n",
      "TCGA-DM-A1DB.h5, 0\n",
      "TCGA-DM-A1HA.h5 missing mutation\n",
      "TCGA-DM-A280.h5, 0\n",
      "TCGA-DM-A282.h5, 0\n",
      "TCGA-DM-A285.h5 missing mutation\n",
      "TCGA-DM-A288.h5, 0\n",
      "TCGA-DM-A28A.h5, 0\n",
      "TCGA-DM-A28E.h5, 0\n",
      "TCGA-DM-A28F.h5 missing mutation\n",
      "TCGA-DM-A28G.h5, 0\n",
      "TCGA-DM-A28H.h5, 0\n",
      "TCGA-DM-A28K.h5, 0\n",
      "TCGA-DM-A28M.h5, 0\n",
      "TCGA-DT-5265.h5 missing mutation\n",
      "TCGA-DY-A0XA.h5, 0\n",
      "TCGA-DY-A1DC.h5, 0\n",
      "TCGA-DY-A1DD.h5 missing mutation\n",
      "TCGA-DY-A1DE.h5, 0\n",
      "TCGA-DY-A1DF.h5 missing mutation\n",
      "TCGA-DY-A1DG.h5 missing mutation\n",
      "TCGA-EF-5830.h5, 0\n",
      "TCGA-EI-6506.h5, 0\n",
      "TCGA-EI-6507.h5, 0\n",
      "TCGA-EI-6882.h5, 0\n",
      "TCGA-EI-6883.h5, 0\n",
      "TCGA-EI-6884.h5, 0\n",
      "TCGA-EI-6885.h5, 1\n",
      "TCGA-EI-6917.h5, 0\n",
      "TCGA-EI-7002.h5, 1\n",
      "TCGA-EI-7004.h5, 1\n",
      "TCGA-F4-6459.h5, 0\n",
      "TCGA-F4-6460.h5, 0\n",
      "TCGA-F4-6461.h5, 0\n",
      "TCGA-F4-6463.h5, 0\n",
      "TCGA-F4-6569.h5, 0\n",
      "TCGA-F4-6570.h5, 0\n",
      "TCGA-F4-6703.h5, 0\n",
      "TCGA-F4-6704.h5, 0\n",
      "TCGA-F4-6805.h5, 0\n",
      "TCGA-F4-6806.h5, 0\n",
      "TCGA-F4-6807.h5, 0\n",
      "TCGA-F4-6808.h5, 0\n",
      "TCGA-F4-6809.h5, 0\n",
      "TCGA-F4-6854.h5, 0\n",
      "TCGA-F4-6855.h5, 0\n",
      "TCGA-F4-6856.h5, 0\n",
      "TCGA-F5-6464.h5, 0\n",
      "TCGA-F5-6465.h5, 0\n",
      "TCGA-F5-6571.h5, 0\n",
      "TCGA-F5-6702.h5, 0\n",
      "TCGA-F5-6810.h5 missing mutation\n",
      "TCGA-F5-6811.h5, 0\n",
      "TCGA-F5-6812.h5, 0\n",
      "TCGA-F5-6813.h5, 0\n",
      "TCGA-F5-6814.h5, 0\n",
      "TCGA-F5-6863.h5, 0\n",
      "TCGA-F5-6864.h5, 0\n",
      "TCGA-G4-6293.h5, 0\n",
      "TCGA-G4-6294.h5, 0\n",
      "TCGA-G4-6295.h5, 0\n",
      "TCGA-G4-6297.h5, 1\n",
      "TCGA-G4-6298.h5, 0\n",
      "TCGA-G4-6299.h5, 1\n",
      "TCGA-G4-6302.h5, 1\n",
      "TCGA-G4-6303.h5, 1\n",
      "TCGA-G4-6317.h5 missing mutation\n",
      "TCGA-G4-6320.h5, 1\n",
      "TCGA-G4-6321.h5, 0\n",
      "TCGA-G4-6322.h5, 1\n",
      "TCGA-G4-6323.h5, 0\n",
      "TCGA-G4-6586.h5, 1\n",
      "TCGA-G4-6588.h5, 0\n",
      "TCGA-G4-6625.h5, 0\n",
      "TCGA-G4-6626.h5, 0\n",
      "TCGA-G4-6627.h5, 0\n",
      "TCGA-G4-6628.h5, 0\n",
      "TCGA-G5-6233.h5, 1\n",
      "TCGA-G5-6235.h5, 0\n",
      "TCGA-G5-6572.h5, 0\n",
      "TCGA-G5-6641.h5, 0\n",
      "TCGA-NH-A50T.h5, 0\n",
      "TCGA-NH-A50U.h5, 0\n",
      "TCGA-NH-A50V.h5, 0\n",
      "TCGA-NH-A5IV.h5, 0\n",
      "TCGA-NH-A6GA.h5, 1\n",
      "TCGA-NH-A6GB.h5, 0\n",
      "TCGA-NH-A6GC.h5, 0\n",
      "TCGA-NH-A8F7.h5, 0\n",
      "TCGA-NH-A8F8.h5, 1\n",
      "TCGA-QG-A5YV.h5, 0\n",
      "TCGA-QG-A5YW.h5, 0\n",
      "TCGA-QG-A5YX.h5, 0\n",
      "TCGA-QG-A5Z1.h5, 1\n",
      "TCGA-QG-A5Z2.h5, 0\n",
      "TCGA-QL-A97D.h5 missing mutation\n",
      "TCGA-SS-A7HO.h5, 1\n",
      "TCGA-T9-A92H.h5, 0\n",
      "TCGA-WS-AB45.h5, 0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Lymphovascular invasion indicator</th>\n",
       "      <th>Filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-3L-AA1B.h5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-3L-AA1B.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-4N-A93T.h5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-4N-A93T.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-4T-AA8H.h5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-4T-AA8H.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-5M-AAT4.h5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-5M-AAT4.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-5M-AAT5.h5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-5M-AAT5.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>488</th>\n",
       "      <td>TCGA-QG-A5Z2.h5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-QG-A5Z2.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489</th>\n",
       "      <td>TCGA-QL-A97D.h5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-QL-A97D.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>TCGA-SS-A7HO.h5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCGA-SS-A7HO.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>TCGA-T9-A92H.h5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-T9-A92H.h5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>TCGA-WS-AB45.h5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-WS-AB45.h5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>493 rows √ó 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Sample_ID  Lymphovascular invasion indicator         Filename\n",
       "0    TCGA-3L-AA1B.h5                                0.0  TCGA-3L-AA1B.h5\n",
       "1    TCGA-4N-A93T.h5                                0.0  TCGA-4N-A93T.h5\n",
       "2    TCGA-4T-AA8H.h5                                0.0  TCGA-4T-AA8H.h5\n",
       "3    TCGA-5M-AAT4.h5                                NaN  TCGA-5M-AAT4.h5\n",
       "4    TCGA-5M-AAT5.h5                                NaN  TCGA-5M-AAT5.h5\n",
       "..               ...                                ...              ...\n",
       "488  TCGA-QG-A5Z2.h5                                0.0  TCGA-QG-A5Z2.h5\n",
       "489  TCGA-QL-A97D.h5                                NaN  TCGA-QL-A97D.h5\n",
       "490  TCGA-SS-A7HO.h5                                1.0  TCGA-SS-A7HO.h5\n",
       "491  TCGA-T9-A92H.h5                                0.0  TCGA-T9-A92H.h5\n",
       "492  TCGA-WS-AB45.h5                                0.0  TCGA-WS-AB45.h5\n",
       "\n",
       "[493 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import h5py\n",
    "import os\n",
    "from glob import glob\n",
    "import pandas as pd\n",
    "\n",
    "h5_paths = sorted(glob(os.path.join(h5_folder, \"*.h5\")))\n",
    "\n",
    "records = []\n",
    "\n",
    "for path in h5_paths:\n",
    "    filename = os.path.basename(path)\n",
    "    sample_id = \"-\".join(filename.split(\"-\")[:3])  # TCGA-XX-YYYY\n",
    "    \n",
    "    try:\n",
    "        with h5py.File(path, \"r\") as f:\n",
    "            if mutation in f:\n",
    "                mmr = f[mutation][()]\n",
    "                print(f\"{sample_id}, {mmr}\")\n",
    "                records.append((sample_id, mmr, filename))\n",
    "            else:\n",
    "                print(f\"{sample_id} missing mutation\")\n",
    "                records.append((sample_id, None, filename))\n",
    "    except Exception as e:\n",
    "        print(f\"Error reading {filename}: {e}\")\n",
    "        records.append((sample_id, None, filename))\n",
    "\n",
    "mutation_read_df = pd.DataFrame(records, columns=[\"Sample_ID\", mutation, \"Filename\"])\n",
    "display(mutation_read_df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (uni-env)",
   "language": "python",
   "name": "uni-env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
